.teamcity/Gradle_Check/model/BucketProvider.kt

```diff
@@ -0,0 +1,239 @@
+package Gradle_Check.model
+
+import com.alibaba.fastjson.JSON
+import com.alibaba.fastjson.JSONArray
+import com.alibaba.fastjson.JSONObject
+import configurations.FunctionalTest
+import model.BuildTypeBucket
+import model.CIBuildModel
+import model.GradleSubProject
+import model.Stage
+import model.TestCoverage
+import model.TestType
+import java.io.File
+
+typealias BuildProjectToSubProjectTestClassTimes = Map<String, Map<String, List<TestClassTime>>>
+
+interface GradleBuildBucketProvider {
```

Comment:
This interface factors out the `FunctionalTest` creation logic so other parts can be greatly simplified.